mantidproject/mantid
# Disable Plot Result for MSD and F(Q) Fit for one spectra

**Description of work.**
This PR ensures that the `Plot Result` button is disabled for a workspace containing only one spectra for both MSD Fit and F(Q) Fit on the Data Analysis Interface of Indirect .
This PR also makes sure an informative error message is displayed when trying to plot a workspace with a single spectra from MultipleInput. The other workspaces with more than one spectra in the MultipleInput will be plotted.
This PR also ensures that the `Plot Result` button on each tab of Data analysis is disabled while plotting is taking place.
The `Calculate Errors` checkbox in the I(Q,t) Tab has been moved to the left of `Number of Iterations`.

**To test:**
1. `Indirect Data Analysis` -> `MSD Fit` Tab (or `F(Q) Fit` Tab)
2. Load the data provided below
3. Selected Plot type `Gaussian`
4. Click `Run`
5. Check that the `Plot Result` (or `Plot`) button is disabled/enabled for the relevant data

- Repeat this but for F(Q) Fit

1. In `MSD Fit`, load both the files below into `Multiple Input` by clicking `Add Workspace`, browsing to the file location and selecting `All Spectra`.
2. Select Plot type `Gaussian` and click `Run`
3. Click `Plot Result` - two graphs should be displayed, and then an informative error message (which corresponds to the last workspace of the two below)

- While plotting is being undertaken, the `Plot Result` button should also be disabled.

**Test Data**
MSD Fit:
[osi104367-104371_graphite002_red_elwin_eq.zip](https://github.com/mantidproject/mantid/files/2352781/osi104367-104371_graphite002_red_elwin_eq.zip) - button should be enabled
[osi92762_graphite002_elwin_eq.zip](https://github.com/mantidproject/mantid/files/2352782/osi92762_graphite002_elwin_eq.zip) - button should be disabled

F(Q) Fit:
[IN16B_125878_QLd_Result.zip](https://github.com/mantidproject/mantid/files/2352784/IN16B_125878_QLd_Result.zip) - button should be disabled
[iris26174_graphite002_s0-17_Result.zip](https://github.com/mantidproject/mantid/files/2379723/iris26174_graphite002_s0-17_Result.zip) - button should be disabled

--------------------------------------------------------

Fixes #23164 and #23496

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
